stdlib/public/core/AutoDiff.swift

```diff
@@ -60,23 +60,25 @@ public extension VectorNumeric {
 public protocol Differentiable {
   /// The tangent vector space of this differentiable manifold.
   associatedtype TangentVector : Differentiable
-    where TangentVector.TangentVector == TangentVector
+    where TangentVector.TangentVector == TangentVector,
+          TangentVector.Scalar : FloatingPoint
```

Comment:
Regarding the build error: I think you need to specify `TangentVector`'s conformance to `VectorNumeric` here.